Sources/Inject/Integrations/SwiftUI.swift

```diff
@@ -2,6 +2,7 @@ import Foundation
 import SwiftUI
```

Comment:
I think you need to guard even imports?

Reply:
No, I don't know what the deployment target for SwiftUI or Combine (or any other of Apple's frameworks) actually is, but apparently it's low enough to be imported anywhere. That's why each type declaration in these frameworks has its own availability statement, like:
```swift
@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
public protocol View { ... }
```
So the frameworks can then be imported anywhere, but the types need to be guarded.

Reply:
use a #if canImport(SwiftUI) ?

Reply:
`canImport` checks for platform availability, not deployment target. From the [documentation](https://docs.swift.org/swift-book/ReferenceManual/Statements.html#ID538):
> The argument for the canImport() platform condition is the name of a module that may not be present on all platforms. This condition tests whether it’s possible to import the module, but doesn’t actually import it. If the module is present, the platform condition returns true; otherwise, it returns false.

So `canImport(SwiftUI)` will be true for all platforms that support SwiftUI.

But like I said, that's not required as Apple's frameworks don't have any deployment targets. So they can always be imported. Only the usage of types needs to be guarded then.

Reply:
I just tried it out in a sample project, where I set the deployment target to iOS 9 (oldest possible with Xcode 13.3.1).

The compiler only complains about missing availability statements. The imports are fine

<img width="771" alt="Bildschirmfoto 2022-05-12 um 18 54 26" src="https://user-images.githubusercontent.com/12034322/168128383-5f37de10-42f5-4e4e-998f-7ba3aaa86f02.png">

Once availability statements are added, the issues are resolved.

<img width="771" alt="Bildschirmfoto 2022-05-12 um 18 54 37" src="https://user-images.githubusercontent.com/12034322/168128568-ec6ac459-a6fd-4961-bc73-4fe1c04bbb80.png">

Reply:
Sweet, thanks for confirming!

Reply:
tvOS, macOS?

Reply:
Doesn't work in practice though - if you run it in an iOS 12 simulator:

```
dyld: Library not loaded: /System/Library/Frameworks/SwiftUI.framework/SwiftUI
  Referenced from: /Users/ian/Library/Developer/CoreSimulator/Devices/C9046621-63BE-4232-A98D-3DCB1FFA227B/data/Containers/Bundle/Application/AA85BE8E-39E6-49B3-A534-4D7FA99801B1/CTR.app/CTR
  Reason: image not found
dyld: launch, loading dependent libraries
```

Reply:
I'm back with [a fix](https://stackoverflow.com/a/57907940/349364):

Just need to add `-weak_framework SwiftUI` to Other Linker Flags, and with that it runs iOS 12 with no issue.